src/GraphQLParser/ROM.cs

```diff
@@ -0,0 +1,100 @@
+using System;
+
+namespace GraphQLParser
+{
+    /// <summary>
+    /// A wrapper around ReadOnlyMemory{char} allowing you to use simple syntax when working with it.
```

Comment:
The main problem with `ReadOnlyMemory{char}` is its `GetHashCode` : 
```c#
 /// <summary>Returns the hash code for this <see cref="ReadOnlyMemory{T}"/></summary>
        [EditorBrowsable(EditorBrowsableState.Never)]
        public override int GetHashCode()
        {
            // We use RuntimeHelpers.GetHashCode instead of Object.GetHashCode because the hash
            // code is based on object identity and referential equality, not deep equality (as common with string).
            return (_object != null) ? HashCode.Combine(RuntimeHelpers.GetHashCode(_object), _index, _length) : 0;
        }
```

My experiments have shown that with such an implementation it is impossible to work with dictionaries, because the hash code for matching pieces of a string will still be different:

example: `abc abc`

`Slice(0, 3).GetHashCode() != Slice(4, 3).GetHashCode()`

In our case I compare **content** of memory chunks.

Reply:
You can specify a comparer for a dictionary to use which overrides the hash code and equality comparisons. https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2.comparer?view=net-5.0

Reply:
I still need ==, != and implicit operators so it is easier to keep all this stuff inside one custom struct.